utils/rjson.hh

```diff
@@ -188,6 +188,8 @@ rjson::value parse_yieldable(chunked_content&&, size_t max_nested_level = defaul
 // Creates a JSON value (of JSON string type) out of internal string representations.
 // The string value is copied, so str's liveness does not need to be persisted.
 rjson::value from_string(const char* str, size_t size);
+rjson::value from_string(const std::string& str);
+rjson::value from_string(const sstring& str);
```

Comment:
> it's just book-keeping in the compiler

I am not sure. It would be great to dump small piece of the code to some kind of IR/SSA or even asm but annotated with c++ code. It's very easy to do in Golang but for C++ I only know how to dump the whole codebase as LLVM bitcode but that's alone not helpful.

What I see during debugging release build is that it enters:
```c++
  operator std::basic_string_view<char_type>() const noexcept {
        ...
        static_assert(noexcept(std::basic_string_view<char_type>(str(), size())));
        return std::basic_string_view<char_type>(str(), size());
    }
```

and as far as I can tell those functions are not inlined nor executed in compile time, so at least 3 function calls and static asserts. Why to slow down compilation and runtime if you can just export all those implemented functions (even if it's very tiny slowdown).

Reply:
C++ is a very slow to compile language, because so much happens at compile time. We can't really fight it, and the compile-time overhead of implicit conversion of std::string or "string literal" to std::string_view  is probably negligible compared to all other compile-time overheads.

One of the reasons std::string_view was introduced was to replace all those uses of `std::string&`, and now you want to bring it back. Bringing back `std::string&` and in our case also `sstring&` is not such a disaster, but I still don't understand where is the benefit :-( It seems to me that it made many of the calls uglier, and a few less efficient. I don't see which of the calls improved its run-time performance by using the `sstring&` overload instead of `std::string_view`.

Reply:
maybe string -> string_view is efficiently  optimized (not sure) but sstring -> string_view calls https://github.com/scylladb/seastar/blob/master/include/seastar/core/sstring.hh#L579 which looks less efficient than just choosing the right overload

in which cases using those extra overloads would be slower, to me seems like callers have full control which overload they want?

to be honest I'd not add those functions if they would not be implemented, when I saw odd stuff like `rjson::from_string(sstring(_collection_name.begin(), _collection_name.end())));` I just though they were accidentally not exposed. If most pieces of the code would use string_view that'd also be a noble quest to try to unify. But to me looks like this attempt (if it was conscious) was slightly artificial as most or many usages are just sstring or string.